### PR TITLE
Add note for necessary step of setting password

### PR DESCRIPTION
**Pull request by FranklinYu**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/39915#issuecomment-439283302

###### Things done

I don’t think they applies since this is only documentation update.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rnhmjoj** [comment]
I would rephrase the note to mention `mutableUsers = false`, and `users.users.passwordFile`, which is probably more frequent use-case than an unattended installation.

Also `sudo nixos-enter --command passwd` seems friendlier than chpasswd.

**Comment by FranklinYu** [comment]
@rnhmjoj Sorry, I’m not sure how these two options are related to this note here. This note is only about the step of running

```console
# nixos-install
```

as well as steps after that. IMO the two configuration options you mention should be specified in the step above it, namely

> You should then edit `/mnt/etc/nixos/configuration.nix` to suit your needs:
>
> ```console
> # nano /mnt/etc/nixos/configuration.nix
> ```

As for the recommendation of `nixos-enter`, I don’t see why we would prefer it over simply *not* using `--no-root-passwd`, since both are intended for the same scenario (interactive installation).

I’m not yet familiar with NixOS so please don’t hesitate to correct my mistakes above.

**Review by Janik-Haag — COMMENTED** [review]
What keeps this from getting merged?

**Comment by Janik-Haag** [comment]
I'll close this for now since no one answered. feel free to reopen anytime you have time to work on it again.
